### PR TITLE
Ret 2956 Provide something Tribunal asked for - Respond to a notification (ECM) - notify selected parties (email)

### PR DESCRIPTION
**Pull request by paul-peasegood**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RET-2956

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by Itay-Kirsh — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/et/common/model/ccd/items/GenericTypeItem.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+@SuperBuilder
+@NoArgsConstructor
+public class GenericTypeItem<T> {
+
+    @JsonProperty("id")
+    private String id;
+    @JsonProperty("value")
+    private T value;
+
+}
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/et/common/model/ccd/items/GenericTypeItem.java:20]: Excellent, hopefully we can use this pattern going forward. Will save a lot of manual work declaring XTypeItems

File: src/main/java/uk/gov/hmcts/et/common/model/ccd/types/RespondNotificationType.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@Data
+@SuperBuilder
+@NoArgsConstructor
+public class RespondNotificationType {
+
+
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/et/common/model/ccd/types/RespondNotificationType.java:18]: double blank line?

File: src/main/java/uk/gov/hmcts/et/common/model/ccd/types/RespondNotificationType.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    @JsonProperty("respondNotificationDate")
+    private String respondNotificationDate;
+    @JsonProperty("respondNotificationTitle")
+    private String respondNotificationTitle;
+    @JsonProperty("respondNotificationAdditionalInfo")
+    private String respondNotificationAdditionalInfo;
+    @JsonProperty("respondNotificationUploadDocument")
+    private List<DocumentTypeItem> respondNotificationUploadDocument;
+    @JsonProperty("respondNotificationIsThis")
+    private String respondNotificationIsThis;
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/et/common/model/ccd/types/RespondNotificationType.java:28]: Might be worth taking the opportunity to rename this, it caused confusion earlier on CaseData

File: src/main/java/uk/gov/hmcts/et/common/model/ccd/types/RespondNotificationType.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    @JsonProperty("respondNotificationWhoRespond")
+    private String respondNotificationWhoRespond;
+    @JsonProperty("respondNotificationCaseManagementMadeBy")
+    private String respondNotificationCaseManagementMadeBy;
+    @JsonProperty("respondNotificationFullName")
+    private String respondNotificationFullName;
+    @JsonProperty("respondNotificationPartyToNotify")
+    private String respondNotificationPartyToNotify;
+
+
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/et/common/model/ccd/types/RespondNotificationType.java:40]: double blank line?

**Review by JackReevies — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/et/common/model/ccd/types/RespondNotificationType.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    @JsonProperty("respondNotificationDate")
+    private String respondNotificationDate;
+    @JsonProperty("respondNotificationTitle")
+    private String respondNotificationTitle;
+    @JsonProperty("respondNotificationAdditionalInfo")
+    private String respondNotificationAdditionalInfo;
+    @JsonProperty("respondNotificationUploadDocument")
+    private List<DocumentTypeItem> respondNotificationUploadDocument;
+    @JsonProperty("respondNotificationCMOorRequest")
+    private String respondNotificationCMOorRequest;
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/et/common/model/ccd/types/RespondNotificationType.java:27]: respondNotificationCmoOrRequest

**Review by JackReevies — APPROVED** [review]

**Review by Itay-Kirsh — APPROVED** [review]

**Pull request merged**